extensions/src/main/java/org/pentaho/platform/web/http/api/resources/ActionResource.java

```diff
@@ -88,14 +86,14 @@ public Response invokeAction(
     @QueryParam( ActionUtil.INVOKER_ACTIONID ) String actionId,
     @QueryParam( ActionUtil.INVOKER_ACTIONCLASS ) String actionClass,
     @QueryParam( ActionUtil.INVOKER_ACTIONUSER ) String actionUser,
-    final String actionParams ) {
+    final ActionParams actionParams ) {
```

Comment:
Yes this will cause the systemListener to fail as it was, although I see you made a change to adapt the listener to this change.   Before merging this please test that the system listener still behaves as expected. A concern with this change is that it forces the caller of this method to know how to create an instance of ActionParams.  The three methods which return actionParams from this class are very confusing.  

**public static ActionParams serialize( final IAction action, final Map<String, Serializable> params ) :** 
a method on the class called serialize which does not return a serialized object but rather an instance of actionParams. 

**public static ActionParams fromJson( final String params ):** 
this method does not accept a normal json string, a json object nor a json file, but rather escaped json which can only be generated from ActionParams.toJson().    I would like a method on this class that can go between normal (unescaped) json and the escaped json need by Jackson so that we don't run into issues with escape characters. 
**empty constructor:** 
I understand this has to be here for Jackson to work correctly but it would return a completely unusable version of ActionParams if called from an outside class. 

All this logic makes it very confusing for the caller of this class to (correctly) create an instance of ActionParams.